tests/pagination/src/commonTest/kotlin/CursorBasedPaginationTest.kt

```diff
@@ -0,0 +1,197 @@
+package pagination
+
+import com.apollographql.apollo3.ApolloClient
+import com.apollographql.apollo3.api.Optional
+import com.apollographql.apollo3.cache.normalized.api.FieldPolicyApolloResolver
+import com.apollographql.apollo3.cache.normalized.api.MemoryCacheFactory
+import com.apollographql.apollo3.cache.normalized.api.MetadataGenerator
+import com.apollographql.apollo3.cache.normalized.api.MetadataGeneratorContext
+import com.apollographql.apollo3.cache.normalized.api.NormalizedCacheFactory
+import com.apollographql.apollo3.cache.normalized.api.Record
+import com.apollographql.apollo3.cache.normalized.api.RecordMerger
+import com.apollographql.apollo3.cache.normalized.api.TypePolicyCacheKeyGenerator
+import com.apollographql.apollo3.cache.normalized.apolloStore
+import com.apollographql.apollo3.cache.normalized.normalizedCache
+import com.apollographql.apollo3.cache.normalized.sql.SqlNormalizedCacheFactory
+import com.apollographql.apollo3.testing.runTest
+import kotlin.test.Test
+import kotlin.test.assertEquals
+
+class CursorPaginationTest {
+  @Test
+  fun cursorBasedMemoryCache() {
+    cursorBased(MemoryCacheFactory())
+  }
+
+  @Test
+  fun cursorBasedBlobSqlCache() {
+    cursorBased(SqlNormalizedCacheFactory(name = "blob", withDates = true))
+  }
+
+  @Test
+  fun cursorBasedJsonSqlCache() {
+    cursorBased(SqlNormalizedCacheFactory(name = "json", withDates = false))
+  }
+
+  private fun cursorBased(cacheFactory: NormalizedCacheFactory) = runTest {
+    val client = ApolloClient.Builder()
+        .normalizedCache(
+            normalizedCacheFactory = cacheFactory,
+            cacheKeyGenerator = TypePolicyCacheKeyGenerator,
+            metadataGenerator = CursorPaginationMetadataGenerator(),
+            apolloResolver = FieldPolicyApolloResolver,
+            recordMerger = CursorPaginationRecordMerger()
+        )
+        .serverUrl("unused")
+        .build()
```

Comment:
Looks like you're not using any of the `ApolloClient` functionality here? If that's the case, you can instanciate an `ApolloStore` directly:

```kolin
ApolloStore(
          normalizedCacheFactory = normalizedCacheFactory,
          cacheKeyGenerator = cacheKeyGenerator,
          metadataGenerator = metadataGenerator,
          apolloResolver = apolloResolver,
          recordMerger = recordMerger
      )
```